backend/src/storage/blob.rs

```diff
@@ -50,59 +155,301 @@ impl StorageProvider for BlobStorageProvider {
             .map_err(|e| {
                 StorageError::DatabaseError(format!("failed to convert to utf8: {}", e))
             })?;
-        println!("s_content == {s_content}");
 
-        let flag: BoolFlag = serde_json::from_str(&s_content)
-            .map_err(|e| StorageError::DatabaseError(format!("failed to parse json: {}", e)))?;
+        let flag: BoolFlag = serde_json::from_str(&s_content).map_err(|e| {
+            StorageError::SerializationError(format!("failed to parse json: {}", e))
+        })?;
 
         Ok(Some(flag))
     }
 
-    async fn create_bool_flag(&self, _flag: BoolFlag) -> Result<(), StorageError> {
-        todo!()
+    async fn create_bool_flag(&self, flag: BoolFlag) -> Result<(), StorageError> {
+        if self.get_bool_flag(&flag.id).await?.is_some() {
+            return Err(StorageError::AlreadyExists);
+        }
+        if self.name_exists(&flag.name).await? {
+            return Err(StorageError::AlreadyExists);
+        }
```

Comment:
kinda feels like this is a duplicate check, I think the metadata file should only be used to look up a flag by name

Reply:
the first check makes sure the id isnt there, but then the second check makes sure the name isnt there. the get_bool_flag method doesnt call the metadata.

Reply:
sync'd with shiv, leaving this in prevents flags with duplicate names being created